src/graphchain/core.py

```diff
@@ -239,31 +264,24 @@ def time_to_result(self, memoize: bool = True) -> float:
             self._time_to_result = total_time
         return total_time
 
-    @property
-    def cache_filename(self) -> str:
-        """Filename of the cache file to load or store."""
-        return f"{self.hash}.joblib.lz4"
-
     def cache_file_exists(self) -> bool:
         """Check if this ``CachedComputation``'s cache file exists."""
-        return self.cache_fs.exists(self.cache_filename)  # type: ignore[attr-defined,no-any-return]
+        return self.cache_fs.exists(f"{self.hash}.time.store")
```

Comment:
This is a workaround for the fact that graphchain no longer knows which file(s) get written to the cache (if any). After calling the (user-supplied) `serialize` function, graphchain will also write the file `f"{self.hash}.time.store"` so this should work just fine and is probably even a bit more robust since we would only assume a cache file (or directory) is available after it has been written to completion.

Graphchain (and this feature) has been compatible with S3 from the beginning so yes, it should continue to work with S3.